### PR TITLE
[feat] add execEnvs

### PR DESCRIPTION
**Pull request by yxxhero**
Signed-off-by: yxxhero <aiopsclub@163.com>

add execEnvs

TODO: in the next major version, remove this function

https://github.com/helmfile/helmfile/discussions/92

### CODE REVIEW

File: pkg/tmpl/context_funcs.go [hunk 1]
```diff
@@ -48,7 +49,8 @@ func (c *Context) createFuncMap() template.FuncMap {
 	return funcMap
 }
 
-func (c *Context) Exec(command string, args []interface{}, inputs ...string) (string, error) {
+// TODO: in the next major version, remove this function.
```
Comment (itscaro) [inline pkg/tmpl/context_funcs.go:52]: Why is there this comment :/

Comment (yxxhero) [reply pkg/tmpl/context_funcs.go:52]: Yes. I think this function can be merged when the main version is changed, that is, exec and execEnv can be merged, which is a destructive modification.

File: pkg/tmpl/context_funcs_test.go [hunk 2]
```diff
@@ -267,3 +267,19 @@ func TestExec(t *testing.T) {
...
+
+// TestExecEnvs tests that ExecEnvs returns the expected output.
+// TODO: in the next major version, this test should be removed.
+func TestExecEnvs(t *testing.T) {
+	ctx := &Context{basePath: "."}
+
+	expected := "foo\n"
+
+	// test that the command is executed with environment variables
+	output, err := ctx.ExecEnvs(map[string]string{"testkey": "foo"}, "bash", []interface{}{"-c", "printenv testkey"}, "")
```
Comment (itscaro) [inline pkg/tmpl/context_funcs_test.go:279]: Can you add a test with `nil` as first arg (for envs) as well? thank you

**Review by itscaro — APPROVED** [review]
LGTM

@mumoshu I merge the PR, feel free to bring it up if you have something in mind

File: README.md [hunk 3]
```diff
@@ -1126,6 +1126,16 @@ mysetting: |
 
 The possibility is endless. Try importing values from your golang app, bash script, jsonnet, or anything!
 
+Then `execEnvs` same as `exec`, but it can receive a dict as the envs.  
+
+A usual usage of `execEnvs` would look like this:  
+
+```yaml
+mysetting: |
+{{ execEnvs (dict "envkey" "envValue") "./mycmd" (list "arg1" "arg2" "--flag1") | indent 2 }}
```
Comment (mumoshu) [inline README.md:1135]: @yxxhero @itscaro I think it's a bit hard to remember the arg ordering! How about naming it `envExec` after the fact that it receives the dict of envvars first, and then command and args in the same manner as `exec`?

Comment (yxxhero) [reply README.md:1135]: ok @mumoshu I change it right row.
